src/CollAction/Controllers/ProjectsController.cs

```diff
@@ -59,13 +61,30 @@ public ViewResult StartInfo()
                 return NotFound();
             }
 
-            var project = await _context.Projects.SingleOrDefaultAsync(m => m.Id == id);
-            if (project == null)
+            List<DisplayProjectViewModel> items = await GetProjectDisplayItemsWhere(p => p.Id == id && p.Status != ProjectStatus.Hidden);
+            if (items.Count == 0)
             {
                 return NotFound();
             }
 
-            return View(project);
+            return View(items.First());
+        }
+
+        public async Task<List<DisplayProjectViewModel>> GetProjectDisplayItemsWhere(Expression<Func<Project, bool>> WhereExpression)
```

Comment:
Does this method belong in the controller? Coming from a rails background I would look to express this somewhere else but if this is an ASP.NET thing to do then that's fine. Don't let it block the pull request.